### PR TITLE
feat: Add Memcached authentication

### PR DESCRIPTION
**Pull request by AJIOB**
~Must be merged only after #2114 and #2116, because this branch contains changes from both PRs.~

### CODE REVIEW

File: docs/Memcached.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Memcached
 
-Set `SCCACHE_MEMCACHED` to a [Memcached](https://memcached.org/) url in format `tcp://<hostname>:<port> ...` to store the cache in a Memcached instance.
+Set `SCCACHE_MEMCACHED_ENDPOINT` to a [Memcached](https://memcached.org/) url in format `tcp://<hostname>:<port> ...` to store the cache in a Memcached instance.
+
+`SCCACHE_MEMCACHED` is a deprecated alias for `SCCACHE_MEMCACHED_ENDPOINT`.
```
Comment (sylvestre) [inline docs/Memcached.md:5]: please write here why it is deprecated

Comment (AJIOB) [reply docs/Memcached.md:5]: Just for variables format unifying. Lots of services uses `SCCACHE_<SERVICE>_ENDPOINT` for setting up the service endpoint to connect with.

Comment (AJIOB) [reply docs/Memcached.md:5]: Do you need that note in the `Memcached.md` file?

**Comment by AJIOB** [comment]
@sylvestre & @Xuanwo, recheck it, please.

I think it is ready to be merged.

**Pull request merged**